tests/Stormpath/Tests/Resource/ApplicationTest.php

```diff
@@ -526,6 +526,42 @@ public function testSave()
         $application->save();
     }
 
+    public function testApiKeyManagement()
+    {
+        $application = self::$application;
+
+        $account = \Stormpath\Resource\Account::instantiate(array(
+            'givenName' => 'Account Name',
+            'surname' => 'Surname',
+            'username' => md5(time()) . 'username',
+            'email' => md5(time()) .'@unknown123.kot',
+            'password' => 'superP4ss'));
+
+        $application->createAccount($account);
+        $account = \Stormpath\Resource\Account::get($account->href);
+
+        $newApiKey = $account->createApiKey();
+        $this->assertNotEmpty($newApiKey->id);
+
+        $apiKey = $application->getApiKey($newApiKey->id);
+        $this->assertEquals($newApiKey, $apiKey);
+
+        $encryptedApiKey = $application->getApiKey($newApiKey->id,
+            array('encryptSecret' => true));
+
+        $this->assertEquals($apiKey->secret, $encryptedApiKey->secret);
+
+        $apiKey->status = 'DISABLED';
+        $apiKey->save();
+        $this->assertEquals('DISABLED', $apiKey->status);
+
+        $apiKey->delete();
+        $apiKey = $application->getApiKey($newApiKey->id);
+        $this->assertNull($apiKey);
+
+        $account->delete();
+    }
```

Comment:
We need to add tests to verify that when an api key is retrieved from the application, it is actually retrieved from the cache. This is something that doesn't look to be implemented at all because this requires that the data store caches the api key when created, when retrieved from the account api key collection (which I think needs to be added) and when retrieved by api key id from the application (this requires a special implementation for cacheing). This last scenario (retrieving the api key from the application) is specially important since for authentication requests we always need to perform this query and we shouldn't need to call the server every time because the api keys never changes, except for when the status changes or it gets deleted at which point the cache should get updated.